smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -417,15 +423,42 @@ pub fn handle_swap_reply(
 ) -> Result<Response, ContractError> {
     // TODO: Remove handling of data. if we keep reply_on_success in the caller function
     match data.clone() {
-        SubMsgResult::Ok(_msg) => handle_swap_success(deps, env, data.try_into()?),
+        SubMsgResult::Ok(msg) => {
+            let resp: Result<MsgSwapExactAmountInResponse, _> = data.try_into();
+            let tokens_out: Result<String, ContractError> = match resp {
+                Ok(msg) => Ok(msg.token_out_amount),
+                Err(_) => {
```

Comment:
I don't fully understand why we are doing the advanced handling here, I believe Cosmwasm doesn't return events from CosmosSdk messages anyway in versions < v2.0. Are we expecting cases where Osmosis returns we are unable to handle the swap response and we don't want to revert the tx in that case?

Reply:
If we're trying to handle different swap types here, we probably want to use an untagged struct instead and deserialize against that